sklearn/utils/_mask.py

```diff
@@ -1,21 +1,40 @@
 import numpy as np
+from scipy import sparse
 
 from . import is_scalar_nan
 from .fixes import _object_dtype_isnan
 
 
-def _get_mask(X, value_to_mask):
+def _get_mask(X, value_to_mask, reconstruct_sparse=False):
     """Compute the boolean mask X == missing_values."""
+    X_ = X.copy()
```

Comment:
why do we need this?

Reply:
The implementation of converting back to sparse matrix was repeated multiple times, so I added it in the `_get_mask` function. Recontruction is being used only when we are precomputing mask for missingindicator. The other places where this function is being called might not pass the entire sparse matrix, but just its data instead. So  to make sure the function does maintain integrity we need to copy `X` in another variable.